Zend/zend_object_handlers.c

```diff
@@ -1674,19 +1674,17 @@ ZEND_API bool zend_check_protected(const zend_class_entry *ce, const zend_class_
 }
 /* }}} */
 
-ZEND_API zend_function *zend_get_call_trampoline_func(const zend_class_entry *ce, zend_string *method_name, bool is_static) /* {{{ */
+ZEND_API ZEND_ATTRIBUTE_NONNULL zend_function *zend_get_call_trampoline_func(
+	const zend_function *magic_call, zend_string *method_name, bool is_static) /* {{{ */
```

Comment:
The `is_static` parameter should also go away then.

Reply:
It is used to add a flag within the function body, and there is no convenient way to extract this information otherwise.

Reply:
You should be able to take this from `magic_call->common.fn_flags` just like the attribute flags.